test/e2e/webdriver/driver.js

```diff
@@ -79,17 +79,22 @@ function wrapElementWithAPI(element, driver) {
     try {
       await element.originalClick();
     } catch (e) {
-      if (
-        e.name === 'ElementClickInterceptedError' &&
-        e.message.includes('<div class="mm-box loading-overlay">')
-      ) {
-        // Wait for the loading overlay to disappear and try again
-        await driver.wait(
-          until.elementIsNotPresent(By.css('.loading-overlay')),
-        );
+      if (e.name === 'ElementClickInterceptedError') {
+        if (e.message.includes('<div class="mm-box loading-overlay">')) {
+          // Wait for the loading overlay to disappear and try again
+          await driver.wait(
+            until.elementIsNotPresent(By.css('.loading-overlay')),
+          );
+        }
+        if (e.message.includes('<div class="modal__backdrop">')) {
+          // Wait for the modal to disappear and try again
+          await driver.wait(
+            until.elementIsNotPresent(By.css('.modal__backdrop')),
```

Comment:
Similar methodology as https://github.com/MetaMask/metamask-extension/pull/22672, we would wait for modal to be completed removed and then perform clicking actions.

PS: Once we completely migrated modal component, we can remove this await.